runtime/compiler/optimizer/HCRGuardAnalysis.cpp

```diff
@@ -191,7 +191,7 @@ void TR_HCRGuardAnalysis::initializeGenAndKillSetInfo()
             }
          }
       else if (ttNode->isTheVirtualGuardForAGuardedInlinedCall()
-               && TR_FearPointAnalysis::virtualGuardsKillFear(comp())
+               && TR_FearPointAnalysis::virtualGuardsKillFear(comp(), ttNode)
```

Comment:
Since we're asking about a particular guard now, I think it would be good to have `virtualGuardsKillFear()` (unconditionally, fatal) assert that it's not looking at an HCR guard. HCR guards are not virtual guards per se, even though they satisfy `isTheVirtualGuardForAGuardedInlinedCall()`... At the very least, we definitely shouldn't be considering allowing an HCR guard to kill fear, since it should instead be generating fear. What do you think of such an assertion?

I *think* the only call site you'd have to adjust is this one, where we'd have to check to make sure it's not an HCR guard before calling `virtualGuardsKillFear()` instead of afterward

Reply:
I am bit confused about this suggestion, so the next part of the code I see checks if the guard kind not `HCRGuard` and then only marks if the guard kills fear. Setting the FATAL Assert here means we never expect to reach this point with HCR Guard. Is this true?

Reply:
I don't think we reach any of the *other* call sites with an HCR guard. For this one, the body of this `else if` clause suggests that we can, but we're not going to do anything with an HCR guard anyway, so we could check first instead, and that would allow for the assertion